provider/clusterrosaclassic/cluster_rosa_classic_resource.go

```diff
@@ -680,13 +689,37 @@ func createClassicClusterObject(ctx context.Context,
 		builder.Version(vBuilder)
 	}
 
-	if state.AdminCredentials != nil {
-		htpasswdUsers := []*cmv1.HTPasswdUserBuilder{}
-		htpasswdUsers = append(htpasswdUsers, cmv1.NewHTPasswdUser().
-			Username(state.AdminCredentials.Username.ValueString()).Password(state.AdminCredentials.Password.ValueString()))
+	if common.HasValue(state.CreateAdminUser) && state.CreateAdminUser.ValueBool() ||
+		common.HasValue(state.AdminCredentials) {
+		if !common.HasValue(state.AdminCredentials) {
+			state.AdminCredentials = flattenAdminCredentials("", "")
+		}
+		username, password := expandAdminCredentials(ctx, state.AdminCredentials, diags)
```

Comment:
```suggestion
	username, password := expandAdminCredentials(ctx, state.AdminCredentials, diags)
	if common.BoolWithFalseDefault(state.CreateAdminUser) || common.HasValue(state.AdminCredentials) {
```
This is refactoring `refactoring-1`
See comment with refactoring `refactoring-2` for the whole suggestion

Reply:
fixed